collector_module/opmon_mongodb_maintenance/create_users.py

```diff
@@ -70,7 +72,12 @@ def _create_admin_users(args, client, passwords):
 
 
 def _create_opmon_users(args, client, passwords):
-    for user, roles in user_roles.items():
+    filtered_user_roles = {
+        key: value for key, value in user_roles.items()
+        if (args.user_to_generate and key == args.user_to_generate)
+        or (not args.user_to_generate and key not in explicit_roles)
+    }
```

Comment:
Maybe an explicit `for` statement would be more readable? Because currently looks a bit nasty 😅

Reply:
Comprehension are short, but readability is a bit off :) Changed to ordinary `for` loop. Not sure that current changes adds additional readability.